### DISCUSSION TITLE
Running on Raspbian Buster OS

### DISCUSSION THREAD

**Discussion by Festerstoybox** [AlexandreRouma/SDRPlusPlus#146] [Q&A]
Has anyone successfully installed and ran this on a pi 3/4 using Raspbian OS? If so would love some pointers as i'm still new to Linux and haven't gotten it working yet.

**Comment by yo2ldk** [comment]
also I want to know if it is possible...
73!

**Reply by Festerstoybox** [reply]
i seen somewhere on this site that someone has it running on pi 4 using the release #2, i would love to know how to get it running on the latest build.

**Reply by perseus68-github** [reply]
I tried today after I verified first that the libraries were in place(libfftw3-dev etc.)
So I gave: sudo dpkg -i sdrpp_debian_buster_amd64.deb
The icon was created but the associated file does contain some error and the program does not start-up.
I'm pretty much stuck 😭

**Reply by AlexandreRouma** [reply]
> Has anyone successfully installed and ran this on a pi 3/4 using Raspbian OS? If so would love some pointers as i'm still new to Linux and haven't gotten it working yet.

It is possible but requires editing a few lines to use OpenGL ES. (I'm pretty sure there's an issue open on this repo explaining exactly how to do it)

> So I gave: sudo dpkg -i sdrpp_debian_buster_amd64.deb

As mentioned in the name of that file, that deb is for amd64 ONLY, not arm32 or aarch64.

**Reply by perseus68-github** [reply]
Hi Alexandre, thanks for flying around 😉. Right now I found #39 and #62. I'll give a look

**Comment by perseus68-github** [comment]
So far I could have it running on my raspberry Pi4 (yeah)

I followed the steps as reported in the past issues (see #39 and #62):
$ mkdir src
$ cd src
$ sudo apt install git
$ git clone https://github.com/AlexandreRouma/SDRPlusPlus
$ cd SDRPlusPlus/
$ sudo apt install build-essential
$ sudo apt install cmake
$ sudo apt install libglew-devel
$ sudo apt install libglfw3-dev
$ sudo apt install libvolk1-dev
$ sudo apt install libfftw3-dev
$ sudo apt install portaudio19-dev
$ sudo apt install libsoapysdr-dev
$ mkdir build
$ cd build
# modified ../CMakeLists.txt, added this line: 
target_link_libraries(sdrpp PRIVATE stdc++fs)
# to allow sdrpp to link, failed without "stdc++fs"
$ cmake ..
$ cd build

then modify ../CMakeLists.txt, added this line: 
target_link_libraries(sdrpp PRIVATE stdc++fs)
to allow sdrpp to link, failed without "stdc++fs"

$ cmake ..
$ make -j 4
...
[ 98%] Linking CXX shared library recorder.so
[ 98%] Built target recorder
[100%] Linking CXX shared library frequency_manager.so
[100%] Built target frequency_manager
pi@raspberrypi:~/src/SDRPlusPlus/build $

pi@raspberrypi:~/src/SDRPlusPlus/build $ sudo make install

Now, when I launch it I have the Open GL problem already found by other users, i.e:

pi@raspberrypi:~/src/SDRPlusPlus/build $ ./sdrpp
[2021-08-04 10:06:22.681] [info] SDR++ v1.0.1
[2021-08-04 10:06:22.682] [warning] Root directory /home/pi/.config/sdrpp does not exist, creating it
[2021-08-04 10:06:22.682] [info] Loading config
[2021-08-04 10:06:22.683] [warning] Config file '/home/pi/.config/sdrpp/config.json' does not exist, creating it
[2021-08-04 10:06:22.684] [warning] ConfigManager locked, waiting...
[2021-08-04 10:06:23.274] [error] Glfw Error 65543: GLX: Failed to create context: GLXBadFBConfig

$ sudo apt-get install mesa-utils
$ glxinfo | grep 'version'
server glx version string: 1.4
client glx version string: 1.4
GLX version: 1.4
    Max core profile version: 0.0
    Max compat profile version: 2.1
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.1
OpenGL version string: 2.1 Mesa 19.3.2
OpenGL shading language version string: 1.20
OpenGL ES profile version string: OpenGL ES 3.1 Mesa 19.3.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
    GL_EXT_separate_shader_objects, GL_EXT_shader_implicit_conversions,

pi@raspberrypi:~/src/SDRPlusPlus/build $ grep -i glfwWindowHint ../src/*.cpp
pi@raspberrypi:~/src/SDRPlusPlus/build $ LIBGL_ALWAYS_SOFTWARE=1 glxinfo | grep 'version'
server glx version string: 1.4
client glx version string: 1.4
GLX version: 1.4
    Max core profile version: 3.3
    Max compat profile version: 3.1
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.1
OpenGL core profile version string: 3.3 (Core Profile) Mesa 19.3.2
OpenGL core profile shading language version string: 3.30
OpenGL version string: 3.1 Mesa 19.3.2
OpenGL shading language version string: 1.40
OpenGL ES profile version string: OpenGL ES 3.1 Mesa 19.3.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
    GL_EXT_shader_implicit_conversions, GL_EXT_shader_integer_mix,

Finally here is the trick to make it work:

Just make the sdrpp command to be preceeded by that option...

pi@raspberrypi:~/src/SDRPlusPlus/build $ LIBGL_ALWAYS_SOFTWARE=1 ./sdrpp

And boom, it works!
I'm still fighting with some module that disdappeared (frequency_managaer) but it works fine

**Comment by perseus68-github** [comment]
Moreover I implemented this suggestion:
![IMG_20210805_150804](https://user-images.githubusercontent.com/84195654/128400256-c2657ebb-452a-4280-9f3f-e93dc3255e0b.jpg)

and I can now launch sdrpp with the simple command:
./sdrpp

**Reply by pmaine** [reply]
I have SDRpp running great on my Raspberry Pi 4B with 8GB of RAM. I am running on the Raspian OS and everything is working great.

Are you still having problems? If yes, I will provide you with the steps I followed.

Whis SDR are you using with SDR++?

**Comment by starvald** [comment]
though i did miss a semicolon off one line
glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_ES_API);
